### PR TITLE
Fixed error handling for channel load failures.

### PR DESCRIPTION
**Pull request by akallabeth**
* Fixes audin channel loading checks, prevents duplicate loading and registration before channel initialized.
* Fixes error handling for channel loading failures

### CODE REVIEW

File: libfreerdp/core/client.c [hunk 1]
```diff
@@ -246,7 +252,10 @@ int freerdp_channels_post_connect(rdpChannels* channels, freerdp* instance)
...
 			pChannelOpenData = &channels->openDataList[index];
 
-			pChannelClientData->pChannelInitEventProc(pChannelClientData->pInitHandle, CHANNEL_EVENT_CONNECTED, hostname, hostnameLength);
+			error = pChannelClientData->pChannelInitEventProc(pChannelClientData->pInitHandle,
+					CHANNEL_EVENT_CONNECTED, hostname, hostnameLength);
+			if (error != CHANNEL_RC_OK)
+				goto fail;
 
 			name = (char*) malloc(9);
 			if (!name)
```
Comment (mfleisz) [inline libfreerdp/core/client.c]: If name is NULL we return a signed value although the function's return type is UINT

File: winpr/include/winpr/wtsapi.h [hunk 2]
```diff
@@ -116,7 +116,7 @@ typedef struct tagCHANNEL_PDU_HEADER
 #define VCEXPORT  __export
 #endif
 
-typedef VOID VCAPITYPE CHANNEL_INIT_EVENT_FN(LPVOID pInitHandle,
+typedef UINT VCAPITYPE CHANNEL_INIT_EVENT_FN(LPVOID pInitHandle,
```
Comment (mfleisz) [inline winpr/include/winpr/wtsapi.h]: Changing this from VOID to UINT breaks compatibility with the Microsoft API: https://msdn.microsoft.com/en-us/library/windows/desktop/aa383568(v=vs.85).aspx
Not sure if we are still aiming for 100% MS API compatibility ..?

Comment (bmiklautz) [inline winpr/include/winpr/wtsapi.h]: @mfleisz  for wtsapi at least basic functionality still aiming for compatibility
@akallabeth would you be so kind and the PR?

**Comment by bmiklautz** [comment]
@akallabeth just re-tested this branch it seems to cause a double free in some circumstances:
```
xfreerdp /v:192.168.50.32 /cert-ignore +clipboard  /drive:tmp,/tmp /sound:sys:alsa /microphone:sys:alsa
```

**Comment by akallabeth** [comment]
@bmiklautz Yes, nasty one. Thx for pointing that out.

**Comment by bmiklautz** [comment]
@akallabeth greate. Thanks for having a look!

**Pull request merged**